File: Data Structures & Algorithms/count-connected-components/submission-1.py
class Solution:
    def countComponents(self, n: int, edges: List[List[int]]) -> int:
        graph = defaultdict(list)
        visited = set()
        count = 0
        for edge in edges:
            graph[edge[0]].append(edge[1])
            graph[edge[1]].append(edge[0])
        for node in range(n):
            if node not in visited:
                self.dfs(node, visited, graph)
                count += 1
        return count
        
    def dfs(self, node: int, visited: set, graph: dict) -> None:
        visited.add(node)
        for neighbor in graph[node]:
            if neighbor not in visited:
                self.dfs(neighbor, visited, graph)

File: Data Structures & Algorithms/number-of-one-bits/submission-1.cs
public class Solution {
    public int HammingWeight(uint n) {
        int count = 0;
        while (n > 0)
        {
            count += (n & 1) == 1 ? 1 : 0;
            n >>= 1;
        }
        return count;
    }
}
